mat64/symmetric.go

```diff
@@ -112,8 +119,12 @@ func (s *SymDense) reuseAs(n int) {
 			Data:   use(s.mat.Data, n*n),
 			Uplo:   blas.Upper,
 		}
+		s.cap = n
 		return
 	}
+	if s.mat.N > s.cap {
```

Comment:
I moved it, but I don't understand why this would never fail. If s.mat.N is non-zero and s.cap is zero this would panic, right?